fetch.bs

```diff
@@ -2353,6 +2457,310 @@ X-Content-Type-Options           = "nosniff" ; case-insensitive</pre>
+ containing a signature (using <var>providedTokenBindingKeyPair</var>) over the
+ <var>providedTokenBindingId</var> as well as <var>tlsConnection</var>'s
+ <a for=connection>token-binding Exported Keying Material</a>, as described in
+ <a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.3">section 3.3</a>
+ of [[!TOKBIND-PROTOCOL]].
+
+ <li><p>Set <var>tokenBindingMessage</var> to a list containing <var>providedTokenBinding</var>
+ as the only <a for=/>token binding</a>.
+
+ <li><p>If <var>httpRequest</var>'s <a for=request>referred-token-binding origin</a> is not null,
+ then run these substeps:
+  <ol>
+   <li><p>Let <var>referredTlsConnection</var> be the result of
+   <a lt="obtain a connection" for=connection>obtaining a connection</a> for
+   <var>httpRequest</var>'s <a for=request>referred-token-binding origin</a>
+   with credentials <code>true</code> from the <a>connection pool</a>.
+
+   <li><p>If <a for=connection>token-binding protocol version</a> in
+   <var>referredTlsConnection</var> is null,
+   set <var>referredTokenBinding</var> to null.
+
+   <li><p>If <a for=connection>token-binding protocol version</a> in
+   <var>referredTlsConnection</var> is not null then:
+    <ol>
+     <li><p>Let <var>referredTokenBindingKeyParameters</var> be
+     <var>referredTlsConnection</var>'s <a for=connection>token-binding key parameters</a>.
+
+     <li><p>Let <var>referredTokenBindingKeyPair</var> be the result of
+     <a abstract-op lt="Get the token-binding key">getting the token-binding key</a>
+     for <var>httpRequest</var>'s <a for=request>referred-token-binding origin</a>
+     and <var>referredTokenBindingKeyParameters</var>.
+
+     <li><p>Let <var>referredTokenBindingId</var> be the result of
+     <a abstract-op lt="Compute a Token Binding ID">computing a token binding ID</a>
+     from <var>referredTokenBindingKeyPair</var> and
+     <var>referredTokenBindingKeyParameters</var>.
+
+     <li><p>Let <var>referredTokenBinding</var> be the result of computing a
+     <a for=/>token binding</a> using <var>referredTokenBindingId</var> and
+     <a for=/>token-binding type</a>
+     <code>referred_token_binding</code>, containing a signature
+     (using <var>referredTokenBindingKeyPair</var>) over the <var>referredTokenBindingId</var>
+     as well as <var>tlsConnection</var>'s
+     <a for=connection>token-binding Exported Keying Material</a>, as described in
+     <a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.3">section 3.3</a>
+     of [[!TOKBIND-PROTOCOL]].
+    </ol>
+
+   <li><p>If <var>referredTokenBinding</var> is not null, <a for=list>append</a>
+   it to <var>tokenBindingMessage</var>.
+  </ol>
+
+  <li><p>If <var>tokenBindingMessage</var> is not null, compute and return its base64url-encoding
+  [[!RFC4648]] as described in
+  <a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3">section 3</a>
+  of [[!TOKBIND-PROTOCOL]]. Otherwise return null.
```

Comment:
What is the definition of a base64url-encoding of a list? Again, I'd defer to TOKBIND-PROTOCOL for this. I don't think that what you've describe here will result in a properly serialized TokenBindingMessage struct.

Reply:
Reworded.